CMakeLists.txt

```diff
@@ -1048,15 +1057,17 @@ set(UA_install_tools_files "tools/generate_datatypes.py"
 install(DIRECTORY ${UA_install_tools_dirs} DESTINATION ${open62541_install_tools_dir} USE_SOURCE_PERMISSIONS)
 install(FILES ${UA_install_tools_files} DESTINATION ${open62541_install_tools_dir})
 
-install(FILES ${PROJECT_BINARY_DIR}/open62541.h DESTINATION include)
-# Export amalgamated header open62541.h which is generated due to build of open62541-object
```

Comment:
I would not change this. The amalgamated header should always be installed. See https://github.com/open62541/open62541/issues/2044#issuecomment-444525735

Reply:
There ought to be no amalgamated open62541.h if you choose to build without that build option, i consider creating it a bug in the first place - or if you follow your comment in #2044 that the amalgamted header is the preferred one, don't install the single ones, having both is just confusing and creates a burden in the long run

Reply:
Also 👍 for no amalgamation for the "proper" install.
It is a good way for easy dissemination. But the header is too large and basically unreadable.

So we should encourage using the real headers if users do a "proper" install and dependencies management is less of a problem. Most users require just "ua_sever.h" or "ua_client.h" anyway.

Ok @pro?

Reply:
Yes, that's also fine for me. So we always install the non-amalgamated version by default, correct? (Or also in the packages, like .deb)

We should then also mention that in the documentation. And maybe even remove the option to install the amalgamated version?

Reply:
Just to understand, what is the rationale against installing a non-amalgamated `open62541.h` in the normal installation process? With non-amalgamated `open62541.h` I mean an header that actually includes all the required headers using the preprocessor (such as "ua_server.h", "ua_client.h", etc). In this way, there would be no need to change the code of the examples when switching between amalgamated and non-amalgamated open62541-installs.

Reply:
See my answer here: https://github.com/open62541/open62541/pull/2296#issuecomment-447328078